docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -0,0 +1,492 @@
+- hardware cannot decode the protobuf sign bytes due to memory constraints.
+
+In an effort to remove Amino from the SDK, a new sign mode needs to be created for hardware devices. [Initial discussions](https://github.com/cosmos/cosmos-sdk/issues/6513) propose a string-based sign mode, which this ADR formally specifies.
+
+## Decision
+
+We propose to have SIGN_MODE_TEXTUAL’s signing payload `SignDocTextual` to be an array of strings. Each string corresponds to one "screen" on the hardware wallet device, with no (or little) additional formatting done by the hardware wallet itself.
+
+```proto
+message SignDocTextual {
+  repeated string screens = 1;
+}
+```
+
+The string array MUST follow the specifications below.
+
+### 1. Bijectivity with Protobuf transactions
+
+The encoding and decoding operations between a Protobuf transaction (whose definition can be found [here](https://github.com/cosmos/cosmos-sdk/blob/master/proto/cosmos/tx/v1beta1/tx.proto#L13)) and the string array MUST be bijective.
+
+We concede that bijectivity is not strictly needed. Avoiding transaction malleability only requires collision resistance on the encoding. Lossless encoding also does not require decodability. However, bijectivity assures both non-malleability and losslessness.
+
+This also prevents users signing over hashed transaction metadata, which is a security concern for Ledger's security team.
+
+We propose to maintain functional tests using bijectivity in the SDK.
+
+### 2. Only ASCII 32-127 characters allowed
+
+Ledger devices have limited character display capabilities, so all strings MUST only contain ASCII characters in the 32-127 range.
+
+In particular, the newline `"\n"` (ASCII: 10) character is forbidden.
+
+### 3. Strings SHOULD have the `<key>: <value>` format
+
+Given the Regex `/^(\* )?(>* )?(.*: )?(.*)$/`, all strings SHOULD match the Regex with capture groups 3 and 4 non-empty. This is helpful for UIs displaying SignDocTextual to users.
+
+- The initial `*` character is optional and denotes the Ledger Expert mode, see #5.
+- Strings can also include a number of `>` character to denote nesting.
+- In the case where the first Regex capture group is not empty, it represents an indicative key, whose associated value is given in the second capture group. This MAY be used in the Ledger app to perform custom on-screen formatting, for example to break long lines into multiple screens.
+
+This Regex is however not mandatory, to allow for some flexibility, for example to display an English sentence to denote end of sections.
+
+The `<value>` itself can contain the `": "` characters.
+
+### 4. Values are encoded using Value Renderers
+
+Value Renderers describe how Protobuf types are encoded to and decoded from a string array. The full specification of Value Renderers can be found in [Annex 1](./adr-050-sign-mode-textual-annex1.md).
+
+### 5. Strings starting with `*` are only shown in Expert mode
```

Comment:
What does Expert mode provide and why is it beneficial to some users? Maybe note why it's worth supporting?

Reply:
added a link to explain expert mode. I didn' find official docs though.